lib/rack/error_handler.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'constants'
+
+module Rack
+  class ErrorHandler
+    def initialize(app)
+      @app = app
+    end
+
+    def log_error(env, error)
+      if error_stream = env[RACK_ERRORS]
+        # Exception#full_message was introduced in Ruby 2.5+. If it's not
+        # available, we fall back to the old behavior of just printing the
+        # exception message.
+        if error.respond_to?(:full_message)
+          error_stream.puts(error.full_message)
+        else
+          error_stream.puts(error.message)
+        end
+      end
+    end
+
+    def bad_request(error)
+      [400, {}, ["Bad Request: #{error.class}"]]
```

Comment:
Apparently the RFCs require some descriptive information. Do we want to ignore that?

Reply:
Maybe for BadRequest, we can allow the class name, since we presumably know all classes, and they are all parameter parsing errors and unlikely to provide sensitive information.  For the 500 error, we shouldn't include the class, as it could be any exception.

Reply:
As I don't really have a strong opinion about it, I've removed it. We can always discuss whether we should add it again in the future.